### ISSUE TITLE
Columns with a header labelOrient set to 'bottom' doesn't work as expected

### ISSUE THREAD

**Issue by nick-trustlab** [altair-viz/altair#2557]
Here's the code to recreate the issue:

```python
plot_df = pd.DataFrame({'severity': {0: 'Mild',
                           1: 'Moderate',
                           2: 'Severe',
                           3: 'Mild',
                           4: 'Moderate',
                           5: 'Severe'},
              'variable': {0: 'vote',
                           1: 'vote',
                           2: 'vote',
                           3: 'actual',
                           4: 'actual',
                           5: 'actual'},
              'value': {0: 0.1,
                        1: 0.3,
                        2: 0.6,
                        3: 0.4,
                        4: 0.3,
                        5: 0.3}})
alt.Chart(plot_df).mark_bar().encode(
    x=alt.X('variable'),
    y=alt.Y('value'),
    color=alt.Color('variable'),
    column=alt.Column('severity', header=alt.Header(titleOrient='bottom', labelOrient='bottom'))
)
```
Produces the following output:
![image](https://user-images.githubusercontent.com/92364287/154429652-f6a2e50c-2334-476b-bb16-ac43e284b971.png)

While the `title` "severity" gets moved to the bottom as expected, the labels (Mild, Moderate, and Severe) are only slightly moved down and overlayed on top of the graph.  Changing the code to use `orient='bottom'` doesn't move either the labels or title down.

I am expecting that both `labelOrient='bottom', titleOrient='bottom'` and `orient='bottom'` to have the same effect and to move the labels and title to the bottom of the graph.

Altair version:  4.2.0
Python version: 3.9.7

---

Please follow these steps to make it more efficient to solve your issue:

- [x] Since Altair is a Python wrapper around the Vega-Lite visualization grammar, [most bugs should be reported directly to Vega-Lite](https://github.com/vega/vega-lite/issues). You can click the Action Button of your Altair chart and "Open in Vega Editor" to create a reproducible Vega-Lite example and see if you get the same error in the Vega Editor.
- [x] Search for duplicate issues.
- [x] Use the latest version of Altair.
- [x] Describe how to reproduce the bug and include the full code and data to reproduce it, ideally using a sample data set from `vega_datasets`.

**Comment by joelostblom** [comment]
This has actually been fixed upstreams in Vega Lite. If you click the three dots action button and then "Open in Vega Editor", you will see the behavior you expect. This will be added to Altair when #2528 is merged and you can install Altair from that PR to try out the behavior now.

**Comment by joelostblom** [comment]
Will keep open until the PR is merged.

**Comment by n49o7** [comment]
A quick-and-dirty workaround is to add `labelPadding = (height of chart) + (font size)`. Then of course it won't display correctly in Vega Editor.

**Comment by joelostblom** [comment]
Fixed in #2528
